### PR TITLE
Clarify the log statements so that it differentiates between sampling…

### PR DESCRIPTION
**Pull request by raymondlam**
… and rate limited.

- Split up RateLimiter.check() to have separate methods for sampling and ratelimiting.
- Updated some test variable namings to differentiate between
RateLimiter/RateLimiterImpl.

### CODE REVIEW

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java [hunk 1]
```diff
@@ -145,10 +121,33 @@ boolean check(PerfMetric metric) {
...
   }
 
+  /**
+   * Check if the {@link PerfMetric} should be sampled. A {@link PerfMetric} is considered sampled
+   * if the device isn't allowed to send the event type and it is not part of a verbose session.
+   *
+   * @param metric {@link PerfMetric} object.
+   * @return true if allowed, false if not allowed.
+   */
+  boolean checkDeviceSampling(PerfMetric metric) {
```
Comment (jeremyjiang-dev) [inline firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java:136]: Nit: Can you rename it as `isDeviceSampled`?

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java [hunk 2]
```diff
@@ -107,34 +105,12 @@ private boolean isDeviceAllowedToSendNetworkEvents() {
...
-      return false;
-    }
-
-    if (metric.hasNetworkRequestMetric()
-        && !isDeviceAllowedToSendNetworkEvents()
-        && !hasVerboseSessions(metric.getNetworkRequestMetric().getPerfSessionsList())) {
-      return false;
-    }
-
+  boolean checkDeviceRateLimiting(PerfMetric metric) {
```
Comment (jeremyjiang-dev) [inline firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java:113]: Nit: Can you rename it as `isDeviceRateLimited`? Methods return boolean are suggested to have names that begin with `is` or `has`.

**Review by jeremyjiang-dev — APPROVED** [review]
Thanks for making the changes!

**Review by visumickey — COMMENTED** [review]
Thanks for doing the change. I see a ton of improvements that needs to be done in the code and it is hard to navigate through that. Appreciate you taking time to swim through this code. Good work!

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java [hunk 3]
```diff
@@ -145,10 +121,33 @@ boolean check(PerfMetric metric) {
     } else if (metric.hasTraceMetric()) {
       return traceLimiter.check(metric);
     } else {
+      // Should not reach here
```
[the remark this answers is not in the document]
Comment (raymondlam) [reply firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java:124]: Gauges are already checked in line 114 with isRateLimitApplicable().

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java [hunk 4]
```diff
@@ -145,10 +121,33 @@ boolean check(PerfMetric metric) {
...
 
+  /**
+   * Check if the {@link PerfMetric} should be sampled. A {@link PerfMetric} is considered sampled
+   * if the device isn't allowed to send the event type and it is not part of a verbose session.
+   *
+   * @param metric {@link PerfMetric} object.
+   * @return true if allowed, false if not allowed.
+   */
+  boolean checkDeviceSampling(PerfMetric metric) {
+    if (metric.hasTraceMetric()
```
[the remark this answers is not in the document]
Comment (raymondlam) [reply firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java:137]: I've made it a bit more clear by grouping the session and device allowed checks.

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java [hunk 5]
```diff
@@ -107,34 +105,12 @@ private boolean isDeviceAllowedToSendNetworkEvents() {
...
-   *   <li>It is a {@link TraceMetric}, the {@link PerfSession} is not verbose and trace metrics are
-   *       sampled.
-   *   <li>It is a {@link NetworkRequestMetric}, the {@link PerfSession} is not verbose and network
-   *       requests are sampled.
-   *   <li>The number of metrics being sent exceeds what the rate limiter allows.
-   * </ul>
+   * Check if the {@link PerfMetric} should be rate limited.
    *
    * @param metric {@link PerfMetric} object.
    * @return true if allowed, false if not allowed.
```
[the remark this answers is not in the document]
Comment (raymondlam) [reply firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java]: The expected result is in the `@return` line.

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java (hunk 1 above)
Comment (raymondlam) [reply firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java]: Done

File: firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java (hunk 2 above)
Comment (raymondlam) [reply firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java]: Done

**Comment by raymondlam** [comment]
@visumickey Can you take a look at the PR again. Thanks!

**Review by visumickey — APPROVED** [review]

**Pull request merged**